alphagov/paas-admin
# [#156923083] Require terms on login

## What

Requires users view and acknowledge our terms before continuing their pazmin session.

## How to run

Quick start (against rafal's env):

```
OAUTH_CLIENT_ID="rafal-login" \
OAUTH_CLIENT_SECRET="secret" \
SERVER_ROOT_URL="http://localhost:3000" \
API_URL="https://api.rafalp.dev.cloudpipeline.digital/" \
BILLING_URL="https://billing.rafalp.dev.cloudpipeline.digital/" \
ACCOUNTS_URL="https://accounts.rafalp.dev.cloudpipeline.digital/" \
ACCOUNTS_SECRET="pe015a478d26f5f95cd" \
NOTIFY_API_KEY="qwerty123456" \
NOTIFY_WELCOME_TEMPLATE_ID="qwerty123456" \
SESSION_SECRET="test_session" \
  npm start
```

## How to set a terms document

Post content to the accounts API

```
curl -u admin:pe015a478d26f5f95cd -H "Content-Type: application/json" -X PUT -d '{"content": "# IMPORTANT THINGS\nAll *your* rights _are_ belong to [us](#link)!"}' https://accounts.rafalp.dev.cloudpipeline.digital/documents/TnC
```

## How to review

* IMPORTANT: require's paas-cf changes before merge
* Check that you are forced to accept any documents before continuing
* Check the experience is acceptable

## Who can review

Not @chrisfarms or @carolinegreen
